src/clj_jargon/metadata.clj

```diff
@@ -38,19 +38,19 @@
    ^String dir-path]
   (validate-path-lengths dir-path)
   (mapv avu2map
-    (if (is-dir? cm dir-path)
-      (.findMetadataValuesForCollection collection-ao dir-path)
-      (.findMetadataValuesForDataObject data-ao dir-path))))
+    (case (object-type cm dir-path)
```

Comment:
This pattern I've used a lot. This case is one where it's not a big improvement, because it's just trading one sort of ObjStat call for another. I'll highlight a case later on where this is actually a performance gain, but I changed it across the board to be consistent.